### PR TITLE
feat: add logger middleware

### PR DESCRIPTION
**Pull request by phchan9**
# This PR contains
- add logger middleware in order to integrate discord bot.
- add `--commandline` options to distinguish the caller is in commandline or bot 
- make this module having declaration files 
- refactor code to encapsulate yargs-related code and make log format smooth

### CODE REVIEW

**Review by KimiWu123 — APPROVED** [review]
not very familiar with ts, but it looks good to me!

File: package.json [hunk 1]
```diff
@@ -39,8 +43,8 @@
         "big.js": "5.2.2",
         "chalk": "2.4.2",
         "ethers": "5.1.3",
-        "luxon": "1.26.0",
         "js-yaml": "4.1.0",
+        "luxon": "^1.27.0",
```
Comment (yurenju) [inline package.json:47]: it would be better if use `npm --save-dev --save-exact` to add the package

**Review by yurenju — APPROVED** [review]
that is awesome!! 🎉🎉🎉

File: package.json (hunk 1 above)
Comment (phchan9) [reply package.json]: fixed.

**Pull request merged**